discord.go

```diff
@@ -96,35 +98,202 @@ func (dm *DiscordManager) SaveServerOptions(oldServerOpts *GlobalServerConfig, n
 	return nil
 }
 
-func (dm *DiscordManager) CommandHandler(s *discordgo.Session, m *discordgo.MessageCreate) {
-	if m.Author.ID == s.State.User.ID {
-		return
+// CommandSessions outputs a full list of all scheduled sessions (P, Q & R), using buildCalendar as a base
+func (dm *DiscordManager) CommandSessions() (string, error) {
+	serverOpts, err := dm.store.LoadServerOptions()
+
+	if err != nil {
+		return "", err
+	}
+
+	start := time.Now()
+	end := start.AddDate(0, 0, 7)
+
+	calendar, err := dm.scheduledRacesManager.buildCalendar(start, end)
+
+	if err != nil {
+		return "", err
+	}
+
+	var msg = fmt.Sprintf("Upcoming sessions on server %s\n", serverOpts.Name)
+
+	for _, event := range calendar {
+		msg += event.Start.Format("Mon, 02 Jan 2006 15:04:05 MST") + "\n"
+		msg += event.Title + "\n"
+		msg += event.Description + "\n\n"
+	}
+
+	return msg, nil
+}
+
+// CommandSchedule outputs an abbreviated list of all scheduled events
+func (dm *DiscordManager) CommandSchedule() (string, error) {
+	serverOpts, err := dm.store.LoadServerOptions()
+
+	if err != nil {
+		return "", err
+	}
+
+	start := time.Now()
+	end := start.AddDate(0, 0, 7)
+	scheduled, err := dm.scheduledRacesManager.getScheduledRaces()
+
+	if err != nil {
+		return "", err
+	}
+
+	var recurring []ScheduledEvent
+
+	for _, scheduledEvent := range scheduled {
+		if scheduledEvent.HasRecurrenceRule() {
+			customRace, ok := scheduledEvent.(*CustomRace)
+
+			if !ok {
+				continue
+			}
+
+			rule, err := customRace.GetRecurrenceRule()
+
+			if err != nil {
+				continue
+			}
+
+			for _, startTime := range rule.Between(start, end, true) {
+				newEvent := *customRace
+				newEvent.Scheduled = startTime
+				newEvent.UUID = uuid.New()
+
+				if customRace.GetScheduledTime() == newEvent.GetScheduledTime() {
+					continue
+				}
+
+				recurring = append(recurring, &newEvent)
+			}
+		}
+	}
+
+	scheduled = append(scheduled, recurring...)
+
+	var msg = fmt.Sprintf("\nUpcoming events on server %s\n\n", serverOpts.Name)
```

Comment:
again i'm being picky:

please can you only use `var` to define a variable inside a function when it is not being assigned immediately. for every other variable, use `:=`.

I like that `var` signifies that the variable exists but doesn't have a value yet. just a little readability thing again :)

Reply:
I'd originally created that as an empty string, then at some point came back and added that initial line to it, forgot to do the := thing